apps/vetrina/src/Vetrina/Main.purs

```diff
@@ -231,14 +235,27 @@ render self =
             , children: [ DOM.text "OK" ]
             }
         ]
-    PurchaseUnexpectedError -> DOM.text "SOMETHING WENT HORRIBLY WRONG SERVER SIDE"
-
-vetrinaContainer :: Array JSX -> JSX
-vetrinaContainer children =
-  DOM.div
-    { className: "vetrina--container"
-    , children
-    }
+    PurchaseUnexpectedError -> vetrinaContainer self $ Array.singleton $
+      Purchase.Error.error
+        { onRetry: onRetry
+        }
+  where
+    onRetry = do
+      Aff.launchAff_ $ tryMagicLogin self
```

Comment:
We don't need to actually login, because we already have the user in the state (and if we don't, the magic login would not help, as in that case we don't have the user credentials in local storage). All we need to do, is just set the `purchaseState` to `NewPurchase` and `accountStatus` according to whatever the `state.user` is.

So something like:
```purescript
case state.user of
  Just user -> self.state { purchaseState = NewPurchase, accountStatus = LoggedInAccount user }
  Nothing   -> self.state { purchaseState = NewPurchase }
```